pkg/minikube/bootstrapper/certs.go

```diff
@@ -95,16 +108,38 @@ func SetupCerts(cmd CommandRunner, k8s KubernetesConfig) error {
 	return nil
 }
 
-func generateCerts(caCert, caKey, pub, priv string, ip net.IP, name string, dnsDomain string) error {
+func generateCerts(
+	caCert, caKey, pub, priv string, ip net.IP, name, dnsDomain string,
+	proxyCACert, proxyCAKey, proxyPub, proxyPriv string,
+) error {
 	if !(util.CanReadFile(caCert) && util.CanReadFile(caKey)) {
 		if err := util.GenerateCACert(caCert, caKey, name); err != nil {
-			return errors.Wrap(err, "Error generating certificate")
+			return errors.Wrap(err, "Error generating CA certificate")
 		}
 	}
 
 	ips := []net.IP{ip, internalIP}
-	if err := util.GenerateSignedCert(pub, priv, ips, util.GetAlternateDNS(dnsDomain), caCert, caKey); err != nil {
+	if err := util.GenerateSignedCert(
+		pub, priv, "minikube", ips, util.GetAlternateDNS(dnsDomain),
```

Comment:
Any reason why we parameterized and hardcoded the machine name here?  I think this should be getting its value from KubernetesConfig.NodeName, which is default "minikube" but can be configured.

Reply:
I didn't want the new cert (the proxy-client cert to be used by the aggregator) to ben named "minikube" (or whatever the machine name was), so I updated the function signature of `GenerateSignedCert (...)` to require that you pass in the subject name...

But it's definitely a mistake on my part that when invoking that modified function, I passed in a hardcoded value. I'll fix this!

Reply:
Actually... taking a second look at this, it was always hardcoded as "minikube". All that's happened here is that that hardcoded value is now an argument _to_ `GenerateSignedCert (...)`, where it was previously hardcoded _in_ `GenerateSignedCert(...)`.

Reply:
Yeah - I actually like that you parameterized it in the function here.  I think it was incorrectly hardcoded in the function before.

Reply:
We don't have a integration test for this, but we should verify that if you specify the `--profile` flag to minikube start, that the certs get generated correctly and everything comes up fine.

Reply:
I'll verify that.

Reply:
@r2d4 The new feature is working as expected even with the `--profile` flag used. As with the apiserver's serving cert, the new cert is created in `~/.minikube` and is common to all minikube VMs.